src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.ts

```diff
@@ -77,30 +89,65 @@ export class HostFormComponent extends CdForm implements OnInit {
     });
   }
 
+  // expand hostnames in case hostname is a pattern
+  checkHostNamePattern(hostname: string) {
+    const braceExpansion = require('brace-expansion');
+    if (hostname.includes('[') && hostname.includes(']') && !hostname.match(/(?![^(]*\)),/g)) {
+      const hostnameRange = hostname
+        .replace(/(?<=\d)\s*-\s*(?=\d)/g, '..')
```

Comment:
Additionally I see lots of regex that are not friendly to interpret:
`/(?![^(]*\)),/g`
`/(?<=\d)\s*-\s*(?=\d)/g`

What about adding a brief explanation nearby?
Additionally, if we are trusting the underlying library (brace-expansion) I'd simply adapt the syntax and let the inner layer do the validation.

So let's check that our charset is there (`,[-]`) and no forbidden chars (`{}`) and perform the[ mapping](https://stackoverflow.com/a/10726800/13173750).